main.py

```diff
@@ -155,60 +164,58 @@ def get_interfaces(self):
+                interface_ids = self._get_interfaces(dpid)
+            else:
+                interface_ids.append(interface_enable_id)
+            for interface_id in interface_ids:
+                switch_id = ":".join(interface_id.split(":")[:-1])
+                interface_number = int(interface_id.split(":")[-1])
+            switch = self.controller.switches[switch_id]
+        except KeyError as exc:
+            return jsonify(f"Switch not found: {exc}"), 404
 
-            try:
-                switch.interfaces[interface_number].enable()
-            except KeyError as exc:
-                error_list.append(f"Switch {switch_id} Interface {exc}")
+        try:
+            switch.interfaces[interface_number].enable()
+        except KeyError as exc:
+            error_list.append(f"Switch {switch_id} Interface {exc}")
 
         if not error_list:
             return jsonify("Operation successful"), 201
         return jsonify({msg_error:
                         error_list}), 409
 
-    @rest('v3/interfaces/disable', methods=['POST'])
+    @rest('v3/interfaces/switch/<dpid>/disable', methods=['POST'])
     @rest('v3/interfaces/<interface_disable_id>/disable', methods=['POST'])
-    def disable_interface(self, interface_disable_id=None):
+    def disable_interface(self, interface_disable_id=None, dpid=None):
         """Administratively disable a list of interfaces in the topology."""
         error_list = []  # List of interfaces that were not deactivated.
         interface_ids = []
         msg_error = "Some interfaces couldn't be found and deactivated: "
-        if not interface_disable_id:
-            interface_ids = self._get_data(request)
-        else:
-            interface_ids.append(interface_disable_id)
-        for interface_id in interface_ids:
-            switch_id = ":".join(interface_id.split(":")[:-1])
-            interface_number = int(interface_id.split(":")[-1])
-
-            try:
-                switch = self.controller.switches[switch_id]
-            except KeyError as exc:
-                return jsonify(f"Switch not found: {exc}"), 404
+        try:
+            if dpid:
+                interface_ids = self._get_interfaces(dpid)
+            else:
+                interface_ids.append(interface_disable_id)
+            for interface_id in interface_ids:
+                switch_id = ":".join(interface_id.split(":")[:-1])
+                interface_number = int(interface_id.split(":")[-1])
+            switch = self.controller.switches[switch_id]
+        except KeyError as exc:
+            return jsonify(f"Switch not found: {exc}"), 404
 
-            try:
-                switch.interfaces[interface_number].disable()
-            except KeyError as exc:
-                error_list.append(f"Switch {switch_id} Interface {exc}")
+        try:
+            switch.interfaces[interface_number].disable()
+        except KeyError as exc:
+            error_list.append(f"Switch {switch_id} Interface {exc}")
```

Comment:
Same changes as in `enable_interface`.

Reply:
All suggested changes have been implemented.